lib/util/git/index.spec.ts

```diff
@@ -258,14 +258,13 @@ describe('util/git/index', () => {
     });
 
     it('should return result even if non-default and not under branchPrefix', async () => {
-      const parentSha = await git.getBranchParentSha('develop');
```

Comment:
dont know what causes it really.

![image](https://user-images.githubusercontent.com/97394622/180788152-41ed56ce-994d-41d2-9486-501f941b87f5.png)

{
  "commands": [
    "rev-parse",
    "e12aaf59d110cd6ac98a548a17f7923c47d5ad0d^"
  ],
  "format": "utf-8"
}
```
"Error: e12aaf59d110cd6ac98a548a17f7923c47d5ad0d^\nfatal: ambiguous argument 'e12aaf59d110cd6ac98a548a17f7923c47d5ad0d^': unknown revision or path not in the working tree.\nUse '--' to separate paths from revisions, like this:\n'git <command> [<revision>...] -- [<file>...]'\n\n    at Object.action (C:\\projects\\ladzaretti-renovate\\node_modules\\simple-git\\src\\lib\\plugins\\error-detection.plugin.ts:38:28)\n    at PluginStore.exec (C:\\projects\\ladzaretti-renovate\\node_modules\\simple-git\\src\\lib\\plugins\\plugin-store.ts:24:29)\n    at C:\\projects\\ladzaretti-renovate\\node_modules\\simple-git\\src\\lib\\runners\\git-executor-chain.ts:114:40\n    at new Promise (<anonymous>)\n    at GitExecutorChain.handleTaskData (C:\\projects\\ladzaretti-renovate\\node_modules\\simple-git\\src\\lib\\runners\\git-executor-chain.ts:111:14)\n    at GitExecutorChain.<anonymous> (C:\\projects\\ladzaretti-renovate\\node_modules\\simple-git\\src\\lib\\runners\\git-executor-chain.ts:88:40)\n    at Generator.next (<anonymous>)\n    at fulfilled (C:\\projects\\ladzaretti-renovate\\node_modules\\simple-git\\dist\\cjs\\index.js:55:24)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)"
```

Reply:
Always returns null in production, or in our tests?

Reply:
in tests, as `develop` is not really part of the project i guess.
but ill test in live just in case

Reply:
ok, so we have the same issue with `getCachedBranchParentShaResult`, it will return an unexpected `undefined` when repoCache is disabled, just as in this pr.
ill open another fix pr i guess

Reply:
https://github.com/renovatebot/renovate/pull/16762

Reply:
Can this PR now be progressed?

Reply:
yes, it ready for review